src/averageAges.js

```diff
@@ -15,11 +16,15 @@
  * @return {number}
  */
 function calculateMenAverageAge(people, century) {
-  // write code here
-  // learn how to use array methods like .filter .map .some .every .find .reduce
-  // avoid using loop and forEach
-  // replace `if ()` statement with &&, || or ?:
-  // without nesting
+  const men = century
+    ? people.filter((p) => p.sex === 'm' && Math.ceil(p.died / 100) === century)
+    : people.filter((p) => p.sex === 'm');
+
+  let sum = 0;
```

Comment:
At this section, I suggest to remove the variable sum and at the line 25, 
``` men.forEach((m) => sum += m.died - m.born); ```
I change forEach for reduce, because this way I don't need to use a auxiliar variable. The code I suggest to be more effective 
``` const sum = men.reduce((acc, m) => acc + (m.died - m.born), 0); ```
 and just verify if the man have length like this
``` return men.length > 0 ? sum / men.length : 0; ```

everything could be like this

```
function calculateMenAverageAge(people, century) {

  const men = century
    ? people.filter((p) => p.sex === 'm' && Math.ceil(p.died / 100) === century)
    : people.filter((p) => p.sex === 'm');
    
    const sum = men.reduce((acc, m) => acc + (m.died - m.born), 0);
    
     return men.length > 0 ? sum / men.length : 0; 
    
    }